pennylane/ops/op_math/linear_combination.py

```diff
@@ -0,0 +1,482 @@
+        return "LinearCombination"
+
+    @property
+    def grouping_indices(self):
+        """Return the grouping indices attribute.
+
+        Returns:
+            list[list[int]]: indices needed to form groups of commuting observables
+        """
+        return self._grouping_indices
+
+    @grouping_indices.setter
+    def grouping_indices(self, value):
+        """Set the grouping indices, if known without explicit computation, or if
+        computation was done externally. The groups are not verified.
+
+        **Example**
+
+        Examples of valid groupings for the LinearCombination
+
+        >>> H = qml.LinearCombination([qml.PauliX('a'), qml.PauliX('b'), qml.PauliY('b')])
+
+        are
+
+        >>> H.grouping_indices = [[0, 1], [2]]
+
+        or
+
+        >>> H.grouping_indices = [[0, 2], [1]]
+
+        since both ``qml.PauliX('a'), qml.PauliX('b')`` and ``qml.PauliX('a'), qml.PauliY('b')`` commute.
+
+
+        Args:
+            value (list[list[int]]): List of lists of indexes of the observables in ``self.ops``. Each sublist
+                represents a group of commuting observables.
+        """
+        if value is None:
+            return
+
+        if (
+            not isinstance(value, Iterable)
+            or any(not isinstance(sublist, Iterable) for sublist in value)
+            or any(i not in range(len(self.ops)) for i in [i for sl in value for i in sl])
+        ):
+            raise ValueError(
+                f"The grouped index value needs to be a tuple of tuples of integers between 0 and the "
+                f"number of observables in the LinearCombination; got {value}"
+            )
+        # make sure all tuples so can be hashable
+        self._grouping_indices = tuple(tuple(sublist) for sublist in value)
+
+    def compute_grouping(self, grouping_type="qwc", method="rlf"):
+        """
+        Compute groups of indices corresponding to commuting observables of this
+        LinearCombination, and store it in the ``grouping_indices`` attribute.
+
+        Args:
+            grouping_type (str): The type of binary relation between Pauli words used to compute the grouping.
+                Can be ``'qwc'``, ``'commuting'``, or ``'anticommuting'``.
+            method (str): The graph coloring heuristic to use in solving minimum clique cover for grouping, which
+                can be ``'lf'`` (Largest First) or ``'rlf'`` (Recursive Largest First).
+        """
+
+        with qml.QueuingManager.stop_recording():
+            self._grouping_indices = _compute_grouping_indices(
+                self.ops, grouping_type=grouping_type, method=method
+            )
+
+    @qml.QueuingManager.stop_recording()
+    def _simplify_coeffs_ops(self, cutoff=1.0e-12):
+        """Simplify coeffs and ops
+
+        Returns:
+            coeffs, ops, pauli_rep"""
+
+        if len(self.ops) == 0:
+            return self
+
+        # try using pauli_rep:
+        if pr := self.pauli_rep:
```

Comment:
So I'm a little confused as to why `self.pauli_rep.operation()` isn't sufficient?

Reply:
Hm good question 🤔 doing that change would likely have further reaching consequences but conceptually it would make sense. Perhaps we defer that as a do-later?